### PR TITLE
libigb: Add RXDCTL.ENABLE bit polling when configuring Rx queues

### PR DESCRIPTION
**Pull request by PawelModrzejewski**
According to I210 datasheet RXDCTL.ENABLE bit shall be polled to confirm
that all operations on the queue are done. This fixes the issue when
sometimes RXDCTL0.ENABLE bit was not set after igb_init() was called.

Reference - I210 datasheet:

  4.5.9 Receive Initialization
  ...
    8. Poll the RXDCTL register until the ENABLE bit is set. The tail should not be bumped before this bit was read as one.
  ...

  4.5.9.2 Dynamic Enabling and Disabling of Receive Queues
  ...
  Disabling a Queue:
  ...
    4. The I210-CS/CL clears RXDCTL.ENABLE only after all pending memory accesses to the descriptor ring or to the buffers are done. The software device driver should poll this bit before releasing the memory allocated to this queue.
  ...

### CODE REVIEW

File: lib/igb.c [hunk 1]
```diff
@@ -615,6 +622,15 @@ igb_reset(struct adapter *adapter)
 			rxdctl |= IGB_RX_HTHRESH << 8;
 			rxdctl |= IGB_RX_WTHRESH << 16;
 			E1000_WRITE_REG(hw, E1000_RXDCTL(i), rxdctl);
+
+			rxdctlpollcnt = 0;
```
Comment (MarcinMiklas) [inline lib/igb.c:622]: Any particular reason why you don't like to use for loop here?

Comment (MarcinMiklas) [reply lib/igb.c:622]: On the other hand do  {} while(); would look nice since you could skip break there.

Comment (PawelModrzejewski) [reply lib/igb.c:622]: I've converted it into _do{} while()_ as suggested, looks a bit better :-)

**Comment by PawelModrzejewski** [comment]
@AVnu-Administration is there someone with merge rights able to review/comment this change?

**Review by MarcinMiklas — APPROVED** [review]

**Comment by PawelModrzejewski** [comment]
Any other comments?

**Comment by PawelModrzejewski** [comment]
IGB_RXDCTL_MAX_POLL define added

**Pull request merged**